tests/workflows/basic_api_node_with_timeout/test_timeout.py

```diff
@@ -0,0 +1,70 @@
+from unittest.mock import Mock, patch
+
+from vellum.workflows.constants import APIRequestMethod
+from vellum.workflows.nodes.displayable.bases.api_node.node import BaseAPINode
+
+
+class TestAPINodeTimeout:
+    """Test timeout functionality in BaseAPINode."""
+
+    def _setup_vellum_mocks(self, node):
+        """Helper method to set up common vellum client mocks."""
+        mock_context = Mock()
+        mock_vellum_client = Mock()
+        mock_vellum_response = Mock()
+        mock_vellum_response.json_ = {"result": "success"}
+        mock_vellum_response.headers = {"content-type": "application/json"}
+        mock_vellum_response.status_code = 200
+        mock_vellum_response.text = '{"result": "success"}'
+        mock_vellum_client.execute_api.return_value = mock_vellum_response
+        mock_context.vellum_client = mock_vellum_client
+        node._context = mock_context
+        return mock_vellum_client, mock_vellum_response
+
+    @patch("vellum.workflows.nodes.displayable.bases.api_node.node.Session")
+    def test_local_execute_api_with_timeout(self, mock_session):
+        """Test that timeout is passed to local requests session."""
+        mock_session_instance = Mock()
+        mock_session.return_value.__enter__.return_value = mock_session_instance
+
+        mock_response = Mock()
+        mock_response.json.return_value = {"result": "success"}
+        mock_response.headers = {"content-type": "application/json"}
+        mock_response.status_code = 200
+        mock_response.text = '{"result": "success"}'
+        mock_session_instance.send.return_value = mock_response
+
+        node: BaseAPINode = BaseAPINode()
+        node.url = "https://example.com"
+
+        node._local_execute_api(
+            data=None,
+            headers={},
+            json={"test": "data"},
+            method=APIRequestMethod.POST,
+            url="https://example.com",
+            timeout=30,
+        )
+
+        # Verify timeout was passed to session.send
+        mock_session_instance.send.assert_called_once()
+        call_args = mock_session_instance.send.call_args
+        assert call_args[1]["timeout"] == 30
+
+    def test_run_method_passes_timeout(self):
+        """Test that the run() method properly passes timeout to _run()."""
+        node: BaseAPINode = BaseAPINode()
+        node.url = "https://example.com"
+        node.method = APIRequestMethod.GET
+        node.timeout = 25
+
+        # Mock the _run method to verify timeout is passed
+        with patch.object(node, "_run") as mock_run:
+            mock_run.return_value = Mock()
+
+            node.run()
+
+            # Verify _run was called with timeout
+            mock_run.assert_called_once_with(
```

Comment:
[nit] if we do the above suggestion, then I think the more informative test in this case would be verifying the vellum API case, where we have an assertion validating the right RequestOptions was created and passed in